### PR TITLE
Improved UX of  creating `TimestampNanosecondArray` with timezones

### LINKED ISSUE #3042: Improved UX of  creating `TimestampNanosecondArray` with timezones
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
While working on https://github.com/apache/arrow-datafusion/pull/4039 and https://github.com/influxdata/influxdb_iox/pull/6061 I ended up having to do this many times:

```rust
Arc::new(TimestampNanosecondArray::from(vec![...],)
  .with_timezone("+0:00".into())
```

While this API is already nicer than previously, there are two more improvements that might help:

**Describe the solution you'd like**
Allow passing `&str` as well as `String` to `with_timezone`:

```rust
Arc::new(TimestampNanosecondArray::from(vec![...],)
  .with_timezone("+0:00")
```

Add an alias for UTC:
```rust
Arc::new(TimestampNanosecondArray::from(vec![...],)
  .with_timezone_utc()
```

**Describe alternatives you've considered**

**Additional context**
cc @waitingkuo  who is actively working to improve the timezone situation in arrow / datafusion ❤️ 

I think this is a good first issue as the code and tests are already there, all that is needed is a signature change (from `String` to `impl Into<String>`) and some tests

### PR DESCRIPTION
**Pull request by src255**
# Which issue does this PR close?
 

Closes #3042
 

# What changes are included in this PR?

The `with_timezone_utc` method is implemented with the timezone `"+00:00"`. Two leading zeros are used for consistency with non-zero offsets. Also, the signature of `with_timezone` has been changed:
```rust
// pub fn with_timezone(&self, timezone: String) -> Self {/* --snip-- */}
pub fn with_timezone(&self, timezone: impl Into<String>) -> Self {/* --snip-- */}
```

# Are there any user-facing changes?

The `with_timezone` now accepts `&str` in addition to `String`. For example,
```rust
let arr1: PrimitiveArray<TimestampMillisecondType> =
    TimestampMillisecondArray::from(vec![
        1546214400000,
        -1546214400000,
    ])
//  .with_timezone("+00:00".to_string());
   .with_timezone("+00:00");

let arr2: PrimitiveArray<TimestampMillisecondType> =
    TimestampMillisecondArray::from(vec![
        1546214400000,
        -1546214400000,
    ])
    .with_timezone_utc();

assert_eq!(
    format!("{:?}", arr1),
    format!("{:?}", arr2)
);
```

### CODE REVIEW

File: arrow-array/src/array/primitive_array.rs [hunk 1]
```diff
@@ -779,8 +779,13 @@ impl<T: ArrowTimestampType> PrimitiveArray<T> {
     }
 
     /// Construct a timestamp array with new timezone
-    pub fn with_timezone(&self, timezone: String) -> Self {
-        self.with_timezone_opt(Some(timezone))
+    pub fn with_timezone(&self, timezone: impl Into<String>) -> Self {
```
Comment (alamb) [inline arrow-array/src/array/primitive_array.rs:782]: ❤️

**Review by alamb — APPROVED** [review]
Looks great -- thank you @src255 

cc @waitingkuo

**Comment by tustvold** [comment]
I took the liberty of resolving the merge conflicts on this

**Comment by src255** [comment]
@tustvold Thank you for resolving the merge conflicts, I really appreciate it!

File: arrow-array/src/array/primitive_array.rs (hunk 1 above)
Comment (waitingkuo) [reply arrow-array/src/array/primitive_array.rs:804]: Thank you @src255 @alamb

**Comment by alamb** [comment]
Great team effort 👍

**Pull request merged**